### ISSUE TITLE
[SMRT Analysis Virtual Machine] Failed in "P_AssembleUnitig/runCaToUnitig"

### ISSUE THREAD

**Issue by blent91** [PacificBiosciences/SMRT-Analysis#168]
Hi,

I installed SMRT analysis VM (guided by https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/SMRT-Analysis-Virtual-Machine-Install) on my HP ProLiant DL360 G6 (484184-B21) server with 16 core and 20GB ram. After first execution of 'vagrant up' to finish installation, I assigned 8 cores and 16GB ram to SMRT analysis VM.

First, I tested my microbial genome assembly. The estimated genome size is 3.2Mb. I used RS_HGAP_assembly.3 to de novo assembly my SMRT cell. It failed at 'runCaToUnitig' as shown in the message: [ERROR] 2015-01-31 06:58:02,307 [smrtpipe.status refreshTargets 413] *** Failed task task://016437/P_AssembleUnitig/runCaToUnitig.
The complete log file is 'microbial_genome_v3.log' shown in zipped attachment.

I tried to troubleshoot (guided by https://github.com/PacificBiosciences/SMRT-Analysis/wiki/SMRT-Portal-Job-Fails), and tested lambda phage SMRT cell in the same assembly protocol (but varied genome size setting from 3.2Mb to 50Kb). 
The lambda test also failed (the complete log file is 'lambda_phage_v3.log').

Following troubleshooting guide, I stepped into 2.b to check whether the failures are to do with the computing modes (single-node or multiple-node).
However, I cannot find environmental variable $SMRT_ROOT in SMRT analysis VM. I guessed the $SMRT_ROOT in SMRT analysis VM might be '/home/vagrant/current.'
I found lines 152-153 of the file 'web.xml' in '/home/vagrant/current/redist/apache-tomcat-7.0.23/webapps/smrtportal/WEB-INF' indicated that my tasks run in single-node mode.

In summary, 

1. Did a distributed RS_Resequencing lambda job succeed? Not tested yet.
2. Did a single-node RS_Resequencing lambda job succeed? No. (The default mode of my SMRT analysis VM seems single-node)
3. Paste the [ERROR] lines from the master.log file. (Showed as https://www.dropbox.com/s/2lx3q3ns3ju4z1y/run_log.zip?dl=0)

Thanks for the help on my issue!

**Comment by mjhsieh** [comment]
Hi, Did you modify the "Vagrantfile" file? Beside modifying smrtpipe.rc file, you need to make sure your vagrant instance running as the spec you want to run. Basically these two lines:
```
    vb.customize ["modifyvm", :id, "--memory", "2048"]
    vb.customize ["modifyvm", :id, "--cpus", "2"]
```

**Comment by blent91** [comment]
Hi,

First I used 8 CPUs and 16Gb ram but all tasks were failed.
After I used 16 CPUs and 32Gb ram, everything works.

I solved it myself but thank you though.

Mengjuei Hsieh <notifications@github.com> 於 2015年2月7日 星期六寫道：

> Hi, Did you modify the "Vagrantfile" file? Beside modifying smrtpipe.rc
> file, you need to make sure your vagrant instance running as the spec you
> want to run. Basically these two lines:
>
>     vb.customize ["modifyvm", :id, "--memory", "2048"]
>     vb.customize ["modifyvm", :id, "--cpus", "2"]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PacificBiosciences/SMRT-Analysis/issues/168#issuecomment-73298711>
> .
>


-- 
Chih-Hsien Yang, Ph.D candidate
Institute of Biomedical Informatics
National Yang Ming University, Taipei, Taiwan 11221
TEL: +886-2-28267316
FAX: +886-2-28206754
d49328002@ym.edu.tw, blent@gel.ym.edu.tw, blent91@gmail.com

**Comment by mjhsieh** [comment]
Dear Chih-Hsien,
    Thanks for reporting back the progress. It's great to hear that the issue has been solved. I am going to close this issue and mark this as solved by customer. :-)
Best,
--mj
